src/materials/czm/CZMMaterialBasePK1.C

```diff
@@ -0,0 +1,336 @@
+        }
+      da_dF = _J * F_itr * Fitr_N_norm + R2temp;
+
+      // compute dBdF
+      // --- compute the derivatitve of dtrace(L)dF
+      // ------start with dDL_dF
+      RankFourTensor dDL_dF;
+      for (unsigned int i = 0; i < 3; i++)
+        for (unsigned int j = 0; j < 3; j++)
+          for (unsigned int l = 0; l < 3; l++)
+            for (unsigned int m = 0; m < 3; m++) {
+              dDL_dF(i, j, l, m) = 0;
+              for (unsigned int k = 0; k < 3; k++)
+                dDL_dF(i, j, l, m) -=
+                    _F_avg_old[_qp](i, k) * dFinv_dF(k, j, l, m);
+            }
+
+      const RankTwoTensor dtraceDL_dF =
+          RijklRij(dDL_dF, RankTwoTensor::Identity());
+
+      // compute the derivatitve of the the second part of dadot_da_avg
+      // (n_i*DL_ij*n_j)
+      for (unsigned int p = 0; p < 3; p++)
+        for (unsigned int q = 0; q < 3; q++) {
+          R2temp(p, q) = 0;
+          for (unsigned int i = 0; i < 3; i++)
+            for (unsigned int j = 0; j < 3; j++)
+              for (unsigned int k = 0; k < 3; k++)
+                for (unsigned int l = 0; l < 3; l++)
+                  R2temp(p, q) += dR_dF(i, k, p, q) * _normals[_qp](k) *
+                                      _DL_avg[_qp](i, j) * _R_avg[_qp](j, l) *
+                                      _normals[_qp](l) +
+                                  dR_dF(j, l, p, q) * _normals[_qp](l) *
+                                      _DL_avg[_qp](i, j) * _R_avg[_qp](i, k) *
+                                      _normals[_qp](k) +
+                                  dDL_dF(i, j, p, q) * _R_avg[_qp](i, k) *
+                                      _normals[_qp](k) * _R_avg[_qp](j, l) *
+                                      _normals[_qp](l);
+        }
+
+      // assemble ddsdotdS_dF*_D
+      dB_dF =
+          _D.outerProduct(dtraceDL_dF - R2temp) + _dadot_da_avg[_qp] * dC_dF;
+    } // end of area cahnges derivatives
+
+    // assemble _dPK1traction_dF as the sum of three contributions: the 1st
+    // terms  includes dadA_dF, the 2nd term includes the contributions of
+    // dB_dF, dC_dF and dD_dF, the 3rd term includes the contritbution of
+    // djump_dF
+
+    //  T1;
+    if (_use_area_change) {
+      Vtemp = (_B + _C) * _traction[_qp] + _D * _traction_inc[_qp];
+      _dPK1traction_dF[_qp] = RjkVi(da_dF, Vtemp);
+    } else
+      _dPK1traction_dF[_qp].zero();
+
+    //  T2;
+    Vtemp = _traction[_qp] + _traction_inc[_qp];
+    RankThreeTensor dT_dF_temp = _a * RijklVj(dC_dF, Vtemp);
+    if (_use_area_change)
+      dT_dF_temp += _a * RijklVj(dB_dF, _traction[_qp]);
+    _dPK1traction_dF[_qp] += dT_dF_temp;
+
+    //  T3;
+    R2temp = (_B + _C + _D) * _dtraction_djump[_qp];
+    Vtemp = _displacement_jump_global[_qp] + _displacement_jump_global_inc[_qp];
+    R3temp = RijklVi(dC_dF, Vtemp);
+    _dPK1traction_dF[_qp] += _a * RijRjkl(R2temp, R3temp);
+  }
+}
+
+void CZMMaterialBasePK1::computeFandL() {
```

Comment:
Do you save anything by doing them both in the same function?  Why not 3 smaller methods:

computeFJ()
computeRU()
computeL()

?